tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -204,6 +205,90 @@ private static void AgentConvertSpan<TInner>(TInner activity, Span span)
             {
                 span.Type = activity5 is null ? SpanTypes.Custom : AgentSpanKind2Type(activity5.Kind, span);
             }
+
+            // extract any ActivityLinks
+            ExtractActivityLinks<TInner>(span, activity5);
+        }
+
+        private static void ExtractActivityLinks<TInner>(Span span, IActivity5? activity5)
+            where TInner : IActivity
+        {
+            if (activity5 is null)
+            {
+                return;
+            }
+
+            foreach (var link in (activity5.Links))
+            {
+                var duckLink = link.DuckCast<IActivityLink>();
+
+                _ = HexString.TryParseTraceId(duckLink!.Context.TraceId.TraceId!, out var newActivityTraceId);
+                _ = HexString.TryParseUInt64(duckLink.Context.SpanId.SpanId!, out var newActivitySpanId);
+                var traceParentSample = duckLink.Context.TraceFlags > 0;
+                var traceState = W3CTraceContextPropagator.ParseTraceState(duckLink.Context.TraceState!);
```

Comment:
Since the compiler yelled at you about `duckLink.Context.TraceState` possibly being null, I think the real fix is to update the `ParseTraceState` to accept a `string?` parameter instead of a `string` parameter. It already handles this scenario by executing `string.IsNullOrWhiteSpace` so this should be fine